aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1221,6 +1235,30 @@ static void addmm_impl_cpu_(
   }
 }
 
+Tensor addmm_zerotensor(const Tensor& self, const Tensor& mat1, const Tensor& mat2, const Scalar& beta, const Scalar& alpha){
+  check_addmm(self, mat1, mat2);
+  const auto self_sizes = self.sizes();
+  Tensor result;
+  // beta*self + alpha*(mat1@mat2) = 0
+  if ((mat1._is_zerotensor() || mat2._is_zerotensor() || alpha.equal(0)) &&
+      (self._is_zerotensor() || beta.equal(0))) {
+    result = at::_efficientzerotensor(self_sizes, self.options());
+    return result;
+  }
+  // beta*self
+  else if (mat1._is_zerotensor() || mat2._is_zerotensor() || alpha.equal(0)){
+    return beta.equal(1) ? self.clone() : self * beta;
+  }
+  // alpha*(mat1@mat2)
+  else if (self._is_zerotensor() || beta.equal(0)){
+    result = at::mm(mat1, mat2);
```

Comment:
Offline discussion:
--> @anjali411 profiled and found this to be the bottleneck

Also `at::mm` basically calls `at::addmm` so there is no speed up..... yet.

Reply:
Actually the dispatcher could just be bypassed with `at::cpu::mm` perhaps, that might fix this.